### PR TITLE
crossEventWorker.go

### PR DESCRIPTION
**Pull request by Tigo9527**

### CODE REVIEW

File: infra/blockchain/eventFetcher.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+func (fetcher EventFetcher) Fetch(block uint64) ([]*vault.VaultCrossRequest, error) {
+	fetcher.FilterOpts.Start = block
+	fetcher.FilterOpts.End = &block
+
+	logIterator, err := fetcher.Filter.FilterCrossRequest(fetcher.FilterOpts, nil, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	defer func() {
```
[the remark this answers is not in the document]
Comment (Tigo9527) [reply infra/blockchain/eventFetcher.go:43]: This way has warning "Unhandled error".

File: infra/blockchain/eventFetcher.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+		_ = logIterator.Close()
+	}()
+
+	var logs []*vault.VaultCrossRequest
+	for {
```
[the remark this answers is not in the document]
Comment (Tigo9527) [reply infra/blockchain/eventFetcher.go:48]: 👍

File: infra/worker/crossEventWorker.go [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+	workers, err := setup()
+	if err != nil {
+		logrus.Error("setup fail")
+		return err
+	}
+	logrus.Info("worker count ", len(workers))
+
+	for _, worker := range workers {
+		logrus.WithField("chain", worker.ChainId()).
+			WithField("vault addr", worker.vaultAddr).
```
[the remark this answers is not in the document]
Comment (Tigo9527) [reply infra/worker/crossEventWorker.go:36]: ok

**Review by Tigo9527 — COMMENTED** [review]
Fixed

**Review by wanliqun — APPROVED** [review]

**Pull request merged**